app/api/discount_codes.py

```diff
@@ -0,0 +1,40 @@
+from flask_rest_jsonapi import ResourceDetail, ResourceList, ResourceRelationship
+from marshmallow_jsonapi.flask import Schema, Relationship
+from marshmallow_jsonapi import fields
+
+from app.api.helpers.permissions import jwt_required
+from app.api.helpers.utilities import dasherize
+from app.models import db
+from app.models.event import Event
+from app.models.user import User
+from app.models.discount_code import DiscountCode
+from app.models.event_invoice import EventInvoice
+
+
+class DiscountCodeSchema(Schema):
+    """
+    API Schema for discount_code Model
+    """
+    class Meta:
+        type_ = 'discount_code'
+        self_view = 'v1.discount_code_detail'
+        self_view_kwargs = {'id': '<id>'}
+        self_view_many = 'v1.discount_code_list'
+        inflect = dasherize
+
+    id = fields.Integer()
+    code = fields.Str()
+    discount_url = fields.Str()
+    value = fields.Float()
+    type = fields.Str()
+    is_active = fields.Boolean()
+
+
+class DiscountCodeRelationship(ResourceRelationship):
+    """
+    Discount Code Relationship
+    """
+    decorators = (jwt_required, )
+    schema = DiscountCodeSchema
+    data_layer = {'session': db.session,
+                  'model': DiscountCode}
```

Comment:
This is because event_invoice has a relationship with discount_codes also in the model. This file is added just for the basic relationship. Further code is to be added. I opened a different issue for that https://github.com/fossasia/open-event-orga-server/issues/3735.